vpr/src/place/centroid_move_generator.h

```diff
@@ -10,10 +11,53 @@
  * This move picks a random block and moves it (swapping with what's there if necessary) to the zero force location
  * It calculates forces/weighs acting on the block based on its connections to other blocks.
  *
+ * The computed centroid location can be optionally biased towards the location of NoC routers
+ * that are reachable from the selected block. A NoC router is reachable from a block if one can
+ * start from the block and reach the NoC router only by traversing low fanout nets. All the blocks
+ * (including NoC routers) that can reach a NoC router form a NoC group.
+ *
  * Returns its choices by filling in affected_blocks.
  */
 class CentroidMoveGenerator : public MoveGenerator {
-    e_create_move propose_move(t_pl_blocks_to_be_moved& blocks_affected, t_propose_action& proposed_action, float rlim, const t_placer_opts& placer_opts, const PlacerCriticalities* /*criticalities*/) override;
+  public:
+    CentroidMoveGenerator();
+    CentroidMoveGenerator(float noc_attraction_weight, size_t high_fanout_net);
```

Comment:
Comment constructors and parameters. First is for a regular CentroidMoveGenerator (only considers netlist connections), while the second is NoC-biased with parameters ...